src/webgpu/api/operation/texture_view/format_reinterpretation.spec.ts

```diff
@@ -0,0 +1,390 @@
+    const reinterpretedResolveView =
+      resolveTexture && resolveTexture.createView({ format: viewFormat });
+
+    // Create a pipeline to blit a src texture to the render attachment.
+    const pipeline = makeBlitPipeline(t.device, viewFormat, {
+      sample: 1,
+      render: sampleCount,
+    });
+
+    // Execute a render pass to sample |sampleSource| into |texture| viewed as |viewFormat|.
+    const commandEncoder = t.device.createCommandEncoder();
+    const pass = commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: reinterpretedRenderView,
+          resolveTarget: reinterpretedResolveView,
+          loadOp: 'load',
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.setPipeline(pipeline);
+    pass.setBindGroup(
+      0,
+      t.device.createBindGroup({
+        layout: pipeline.getBindGroupLayout(0),
+        entries: [
+          {
+            binding: 0,
+            resource: sampleSource.createView(),
+          },
+        ],
+      })
+    );
+    pass.draw(6);
+    pass.end();
+
+    // If the render target is multisampled, we'll manually resolve it to check
+    // the contents.
+    const singleSampleRenderTexture = resolveTexture
+      ? t.device.createTexture({
+          format,
+          size: [kTextureSize, kTextureSize],
+          usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+        })
+      : renderTexture;
+
+    if (resolveTexture) {
+      // Create a pipeline to blit the multisampled render texture to a non-multisample texture.
+      // We are basically performing a manual resolve step to the same format as the original
+      // render texture to check its contents.
+      const pipeline = makeBlitPipeline(t.device, format, { sample: sampleCount, render: 1 });
+      const pass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: singleSampleRenderTexture.createView(),
+            loadOp: 'load',
+            storeOp: 'store',
+          },
+        ],
+      });
+      pass.setPipeline(pipeline);
+      pass.setBindGroup(
+        0,
+        t.device.createBindGroup({
+          layout: pipeline.getBindGroupLayout(0),
+          entries: [
+            {
+              binding: 0,
+              resource: renderTexture.createView(),
+            },
+          ],
+        })
+      );
+      pass.draw(6);
+      pass.end();
+    }
+
+    // Submit the commands.
+    t.device.queue.submit([commandEncoder.finish()]);
+
+    // Check the rendered contents.
+    const renderViewTexels = TexelView.fromTexelsAsColors(viewFormat, inputTexelView.color, {
+      clampToFormatRange: true,
+    });
+    t.expectOK(
+      await textureContentIsOKByT2B(
+        t,
+        { texture: singleSampleRenderTexture },
+        [kTextureSize, kTextureSize],
+        { expTexelView: renderViewTexels },
+        { maxDiffULPsForNormFormat: 2 }
```

Comment:
curious, do you happen to know why 2 is needed here?

Reply:
because the resolve operation blits into the resolve texture. On Vulkan, the exact values here can vary.